### PR TITLE
feat(ui): DOMA-5616 stepper

### PR DESCRIPTION
**Pull request by SavelevMatthew**
https://user-images.githubusercontent.com/50069872/228192086-ebb1a9b0-214c-4819-83eb-4ab3c0855bd7.mov

### CODE REVIEW

**Comment by Alllex202** [comment]
what is stepper? :)
add demo please

**Review by pahaz — APPROVED** [review]

File: packages/ui/src/components/Steps/steps.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    // [X . . .]
+    useEffect(() => {
+        const maxAllowedSlide = totalItems - stepsToShow
+        carouselRef.current?.goTo(Math.min(maxAllowedSlide, currentItem))
+    }, [currentItem, totalItems, stepsToShow])
+
+    useEffect(() => {
+        if (onChange) {
+            onChange(currentItem)
+        }
```
Comment (AleX83Xpert) [inline packages/ui/src/components/Steps/steps.tsx:56]: Am I understood right that steps do not contain a body and we should render the step body separately depending on currentStepIndex?

**Review by AleX83Xpert — COMMENTED** [review]
It s good to provide a couple of screenshots.

File: packages/ui/src/components/Steps/step.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        [STEP_CLASS_PREFIX]: true,
+        [`${STEP_CLASS_PREFIX}-${size}`]: size,
+        [`${STEP_CLASS_PREFIX}-${type}`]: type,
+    })
+
+    const indexLevel = size == 'large' ? 4 : 5
+    const indexType = type == 'disabled' ? 'secondary' : 'inverted'
+    const iconSize = size == 'large' ? 'large' : 'medium'
+    const titleSize = size == 'large' ? 'large' : 'medium'
+    const titleType = type == 'disabled' ? 'secondary' : undefined
```
Comment (Alllex202) [inline packages/ui/src/components/Steps/step.tsx:32]: Why do you use `==` and not `===`?
```suggestion
    const indexLevel = size === 'large' ? 4 : 5
    const indexType = type === 'disabled' ? 'secondary' : 'inverted'
    const iconSize = size === 'large' ? 'large' : 'medium'
    const titleSize = size === 'large' ? 'large' : 'medium'
    const titleType = type === 'disabled' ? 'secondary' : undefined
```

**Review by Alllex202 — APPROVED** [review]

File: packages/ui/src/components/Carousel/carousel.tsx [hunk 3]
```diff
@@ -27,19 +22,23 @@ export type CarouselProps = Pick<DefaultCarouselProps,
 | 'effect'
 > & {
     dots?: boolean
+    controlsSize?: 'large' | 'small'
```
Comment (nomerdvadcatpyat) [inline packages/ui/src/components/Carousel/carousel.tsx:25]: Do we need a medium size?

Comment (SavelevMatthew) [reply packages/ui/src/components/Carousel/carousel.tsx:25]: according to design team, we do not

File: packages/ui/src/components/Steps/steps.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+const STEP_GAP = 24
+const STEP_LG_MAX_WIDTH = 40 + 12 + 220 // index circle + gap + text
+const STEP_SM_MAX_WIDTH = 32 + 12 + 200 // index circle + gap + text
+
+const STEPS_CLASS_PREFIX = 'condo-steps'
+
+export const Steps: React.FC<StepsProps> = ({ items, current = 0, onChange, itemSize = 'large' }) => {
+    // NOTE: Internal state controls view
+    const [currentItem, setCurrentItem] = useState(current)
```
[the remark this answers is not in the document]
Comment (SavelevMatthew) [reply packages/ui/src/components/Steps/steps.tsx]: Changed to `currentStep`

**Review by AleX83Xpert — APPROVED** [review]

**Pull request merged**